### PR TITLE
Fix target_info labels

### PR DESCRIPTION
**Pull request by ie-pham**
**What this PR does**: The `labels` variable in the gauge & counter `collectMetrics` function was created outside the for loop and so we were constantly adding labels to metrics after metrics. 

**Which issue(s) this PR fixes**:
Fixes #<issue number>

**Checklist**
- [x] Tests updated
- [ ] Documentation added
- [x] `CHANGELOG.md` updated - the order of entries should be `[CHANGE]`, `[FEATURE]`, `[ENHANCEMENT]`, `[BUGFIX]`

### CODE REVIEW

File: modules/generator/registry/gauge.go [hunk 1]
```diff
@@ -118,31 +118,27 @@ func (g *gauge) collectMetrics(appender storage.Appender, timeMs int64, external
...
-	// set metric name
-	lb.Set(labels.MetricName, g.metricName)
-	// set external labels
-	for name, value := range externalLabels {
-		lb.Set(name, value)
-	}
-
 	for _, s := range g.series {
 		t := time.UnixMilli(timeMs)
+		lbls := make(labels.Labels, 1+len(externalLabels)+len(s.labels.names))
```
Comment (zalegrala) [inline modules/generator/registry/gauge.go:123]: We might want to keep the allocations outside of the loop.  I think I saw a `Reset()` method on the lb object that might empty this out.

Comment (ie-pham) [reply modules/generator/registry/gauge.go]: good call

**Review by calebamiles — APPROVED** [review]
🚀

**Review by mapno — COMMENTED** [review]
I think it's missing a test to ensure the bug is fixed. Otherwise, LGTM

**Comment by ie-pham** [comment]
added test and made sure they failed before the fix and passed after the fix

**Review by mapno — APPROVED** [review]
LGMT, nice fix

**Pull request merged**